lib/adapter.js

```diff
@@ -2481,18 +2481,21 @@ function Adapter(options) {
                 if (typeof _enumList === 'string') {
                     _enumList = [_enumList];
                 }
-                let count = 0;
+                const promises = [];
+                let error;
                 for (let t = 0; t < _enumList.length; t++) {
-                    count++;
-                    this.getEnum(_enumList[t], options, (err, list, _enum) => {
-                        if (list) {
-                            _enums[_enum] = list;
-                        }
-                        if (!--count && callback) {
-                            callback(err, _enums);
-                        }
-                    });
+                    promises.push(new Promise(resolve => {
+                        this.getEnum(_enumList[t], options, (err, list, _enum) => {
+                            if (list) {
+                                _enums[_enum] = list;
+                            }
+                            // we save an error if one occurs
+                            error = err || error;
+                            resolve();
+                        });
+                    }));
                 }
+                Promise.all(promises).then(() => typeof callback === 'function' && callback(error, _enums));
```

Comment:
I have some questions regarding the intended behavior:
* Is the callback supposed to be called with an error and some enums in case there was an error? (Thats how it is now)
* Or should we abort the entire procedure if one request fails?
* Is it okay if all the getEnum requests are fired at once (how it is now) or should we work the requests sequentially?

Reply:
Everywhere possible, I tried to stick to the existing behaviour, and this is how it behaved until now. However, if we touch it now, we can also change the behaviour if desired.

Reply:
Ok is the existing behavior then the first bullet point? I.e. an error plus some enums is possible? That doesn't seem to be in line with many other methods.

Reply:
Yes, it is as shown here: https://github.com/ioBroker/ioBroker.js-controller/pull/747/files/cd12e1c84d33eca7c62bdba4f17fd1b9578ea875#diff-a7bcb5fcc46018435e1ce3fa65fcb992L2485-L2494

Reply:
I will change it to returning the error immediately w.o. partial results. As mentioned by @AlCalzone this will be more consistent with the behaviour of existing methods. 
@Apollon77